python/foglamp/services/core/api/notification.py

```diff
@@ -0,0 +1,517 @@
+                "notification_type": notification_config['notification_type']['value'],
+                "enable": notification_config['enable']['value'],
+            }
+    except Exception as ex:
+        raise web.HTTPInternalServerError(reason=ex)
+    else:
+        return web.json_response({'notification': notification})
+
+
+async def get_notifications(request):
+    """ GET list of notifications
+
+    :Example:
+        curl -X GET http://localhost:8081/foglamp/notification
+    """
+    try:
+        storage = connect.get_storage_async()
+        config_mgr = ConfigurationManager(storage)
+        all_notifications = await config_mgr._read_all_child_category_names("Notifications")
+        notifications = []
+        for notification in all_notifications:
+            notification_config = await config_mgr._read_category_val(notification['child'])
+            notification = {
+                "name": notification_config['name']['value'],
+                "rule": notification_config['rule']['value'],
+                "channel": notification_config['channel']['value'],
+                "notification_type": notification_config['notification_type']['value'],
+                "enable": notification_config['enable']['value'],
+            }
+            notifications.append(notification)
+    except Exception as ex:
+        raise web.HTTPInternalServerError(reason=ex)
+    else:
+        return web.json_response({'notifications': notifications})
+
+
+async def post_notification(request):
+    """
+    Create a new notification to run a specific plugin
+
+    :Example:
+             curl -X POST http://localhost:8081/foglamp/notification -d '{"name": "Test Notification", "description":"Test Notification", "rule": "threshold", "channel": "email", "notification_type": "one shot", "enabled": false}'
+             curl -X POST http://localhost:8081/foglamp/notification -d '{"name": "Test Notification", "description":"Test Notification", "rule": "threshold", "channel": "email", "notification_type": "one shot", "enabled": false, "ruleConfig": {}, "deliveryConfig": {}}'
+    """
+    try:
+        notification_service = ServiceRegistry.get(s_type=ServiceRecord.Type.Notification.name)
+        _address, _port = notification_service[0]._address, notification_service[0]._management_port
+    except service_registry_exceptions.DoesNotExist:
+        return {'result': "No Notification service available."}
+
+    try:
+        data = await request.json()
+        if not isinstance(data, dict):
+            raise ValueError('Data payload must be a valid JSON')
+
+        name = data.get('name', None)
+        description = data.get('description', None)
+        rule = data.get('rule', None)
+        channel = data.get('channel', None)
+        notification_type = data.get('notification_type', None)
+        enabled = data.get('enabled', None)
+        rule_config = data.get('ruleConfig', {})
+        delivery_config = data.get('deliveryConfig', {})
```

Comment:
Please use one syntax `snake_case`, as we used same convention for other endpoints
```
data.get('rule_config', {})
data.get('delivery_config', {})
```

Reply:
done